sklearn/metrics/tests/test_pairwise.py

```diff
@@ -384,6 +384,15 @@ def test_euclidean_distances():
     assert_array_almost_equal(D, [[1., 2.]])
 
     rng = np.random.RandomState(0)
+    #check if it works for float32
+    X = rng.rand(1,3000000).astype(np.float32)
```

Comment:
Is there a specific reason for such a complex test? It seems that the current problem can be reproduced with very simple array, e,g
```python
arr1_32 = np.array([555.5, 666.6, 777.7], dtype=np.float32)
arr2_32 = np.array([555.6, 666.7, 777.8], dtype=np.float32)
arr1_64 = np.array([555.5, 666.6, 777.7], dtype=np.float64)
arr2_64 = np.array([555.6, 666.7, 777.8], dtype=np.float64)
euclidean_distances(arr1_32.reshape(1, -1), arr2_32.reshape(1, -1))
# array([[ 0.17319804]], dtype=float32)
euclidean_distances(arr1_64.reshape(1, -1), arr2_64.reshape(1, -1))
# array([[ 0.17320508]])
```

Reply:
Well I thought it would be better to show that it fails with a random array. No particular reason though. I'll change it to a simpler test if it's a problem.